datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -579,6 +579,44 @@ pub fn array_pop_back(args: &[ArrayRef]) -> Result<ArrayRef> {
     )
 }
 
+pub fn gen_range(args: &[ArrayRef]) -> Result<ArrayRef> {
```

Comment:
This is a very nice implementation 👨‍🍳 👌 

Can you please add a docstring that explains that the arguments are and how they are interpreted

Perhaps following the great example of @jayzhan211  in https://github.com/apache/arrow-datafusion/blob/93a95775f51d21445958067b1da0991879464bb9/datafusion/physical-expr/src/array_expressions.rs#L582-L600